api/src/main/java/com/velocitypowered/api/util/GameProfile.java

```diff
@@ -36,6 +48,36 @@ public String getName() {
     return properties;
   }
 
+  public GameProfile setUniqueId(UUID id) {
```

Comment:
There is no Javadoc here, but more importantly these names are misleading since the class is immutable and `set<something>` is typically used for mutable objects. Prefer `with<something>` instead.